docs/software/create-vo-client.md

```diff
@@ -1,108 +1,168 @@
 
--   start with a clean directory
--   copy in the `vomses`, `gums.template`, and `edg-mkgridmap.conf` files and rename the edg-mkgridmap file:
+The JIRA ticket should contain the details of the changes expected in the new VO Client Package release, which you
+should verify before proceeding.
+You can verify this with your favorite git tool (eg, `git diff` or `gitk`), or just view the changes directly on GitHub:
 
-<!-- -->
+-   <https://github.com/opensciencegrid/osg-vo-config/compare/release-84...master>
 
-          wget http://repo.opensciencgrid.org/pacman/tarballs/vo-package/vomses
-          wget http://repo.opensciencegrid.org/pacman/tarballs/vo-package/edg-mkgridmap.osg
-          mv edg-mkgridmap.osg edg-mkgridmap.conf
-          wget http://repo.opensciencegrid.org/pacman/tarballs/vo-package/gums.template # TODO UNSURE
+Here, `release-84` is the _previous_ release tag, which you are comparing to the latest changes in `master`.
+To use GitHub to view the comparison, you need to specify whatever is the most recent _previous_ release tag.
 
--   Create the vomsdir directory by downloading the .lsc files
+Alternatively, you can proceed to [make the tarball](#making-the-tarball), and compare the result to the previous
+`vo-client` tarball (from the upstream source cache) before [publishing the new release](#publishing-the-new-release).
 
-<!-- -->
+However you choose to do it, the point is to verify that the changes going into the release match what is expected in
+the JIRA ticket before publishing a new release.
 
-         wget --recursive --no-host-directories --cut-dirs=3 -A "*.lsc" http://repo.opensciencegrid.org/pacman/tarballs/vo-package/vomsdir
 
--   In a separate directory, unpack the *old* vo-client tarball (from the upstream source cache)
--   diff the two directories, and compare the changes to the expected changes listed in the JIRA ticket for this VO Client package release
+Updates to the GUMS Template
+----------------------------
 
-<!-- -->
+Most commonly, VO Client Package releases do not involve changes to the `gums.config.template` file, though on occasion
+it needs to be updated.
+Before proceeding, any changes to `gums.config.template` related to this release should be committed to git and pushed
+to the upstream repo on GitHub.
 
--   Follow the instructions in the attached [gums-template-conversion.txt](gums-template-conversion.txt) file to convert it from GUMS 1.1 (1.2?) format to GUMS 1.3 format. Name the result `gums.config.template`. See also the [Automated GUMS Conversion](#automated-gums-conversion) section below for a scripted version of this step.
--   Move the files into a subdirectory to include in the tarball:
+The procedure for updating `gums.config.template` is outside the scope of this document, but the main important point is
+that any updates to this file should be done with the GUMS web interface rather than editing its xml contents by hand.
```

Comment:
But isn't this what needs to get updated to make updates to the VO mapfiles?

Reply:
Yeah, it's true that `voms-mapfile-default` and `grid-vorolemap` are generated based on `gums.config.template`, so what's required is that `gums.config.template` be up to date in git.

The way I see it, this document is for **the release process** for the `vo-client` package, and updates to `gums.config.template` are part of the software development, as opposed to release.

The reason why there used to be an "Automated GUMS Conversion" section is because it used to be a required step in the release process to convert the final, published, old-format `gums.template` from the GOC to the new format using a weaponized GUMS instance.  (And it _used_ to be even more complicated before that.)  But still this did not cover making changes to the gums template (the "development work") -- really it was just "part of the build".

(And you thought the gratia-probe build was absurd.)

The point being, I think any discussion about "Making Changes to the GUMS Template" belongs in a separate document, which we can link to from here.

Reply:
> Right, we usually implement all that stuff in terms of `vomses` and the `.lsc` files, we haven't been touching the GUMS config in some time.

But if we drop or add VOs, don't we need to make sure that we update the mapping?

> make a new VO Package Maintenance Guide

If by this you mean changes to `vomses`, `.lsc`, and `gums.config.template` then yeah, I'm on board.

Reply:
> But if we drop or add VOs, don't we need to make sure that we update the mapping?

I think that's true, but nearly all the updates in the last year have involved only vomses/lsc files.  When we make the new document, we'll have to nail down exactly what needs to be done for what.

And as a side note, for as long as we rely on the gums template for generating the other files, we may want to consider making a tool to ensure consistency between the gums-template-based outputs and the vomses/lsc files.

> > make a new VO Package Maintenance Guide
> 
> If by this you mean changes to `vomses`, `.lsc`, and `gums.config.template` then yeah, I'm on board.

Yes but more generally I mean changes to any of the "data" files in the upstream `osg-vo-config` source repo, which currently includes the items you mentioned + the edg-mkgridmap config, but in the future i suppose the set of files there may change.  (I would also consider including any relevant special details relating to packaging in a Maintenance Guide, but i don't think there are any.)